### ISSUE TITLE
"Extract to function": rename doesn't always works

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/TypeScript#41751]
**TypeScript Version:**  4.1.2

**Search Terms:**

**Code**

```tsx
const App = () => (
    <div>
        <div>foo</div>
    </div>
);
```

1. Highlight the line with `<div>foo</div>`
1. Run "extract to function in module scope"

**Expected behavior:**
- The function is extracted and the user is prompted to rename the new function.

**Actual behavior:**
- The function is extracted and but the user is _not_ prompted to rename the new function. Instead it says "you cannot rename this element".

![image](https://user-images.githubusercontent.com/921609/100758714-b9a51d80-33e7-11eb-8b04-19fc44a2070c.png)

**Playground Link:** 

**Related Issues:**

**Comment by a-tarasyuk** [comment]
Are there additional steps to reproduce the issue? It seems extraction works as expected. Also, there is _fourslash_ test to cover a similar case 

https://github.com/microsoft/TypeScript/blob/33e7c729b9d6881a4f87af6b1c31e428d9e1b05d/tests/cases/fourslash/extract-method_jsxElement1.ts#L3-L32

![bug_41751](https://user-images.githubusercontent.com/509265/101533228-3d3dac00-399e-11eb-9c19-3d0999c75fb1.gif)

**Comment by OliverJAsh** [comment]
That's really odd. Here are my _exact_ steps to reproduce.

1. Clone https://github.com/OliverJAsh/ts-extract-function-jsx-bug
2. `cd` into the cloned repo
3. Run `code --disable-extensions --user-data-dir temp .` (just to make sure no settings/extensions are interfering)
4. Navigate to `main.tsx`
5. Try to extract as per above

![gif](https://user-images.githubusercontent.com/921609/101536581-1b3e2c80-3992-11eb-9c5e-0b81dd3c2436.gif)

**Comment by a-tarasyuk** [comment]
Can you reproduce this issue with the latest TS version? You can change it in VSCode.

**Comment by OliverJAsh** [comment]
Yes, I just tried. Here is a new screenshot—see TS version at the bottom.

![image](https://user-images.githubusercontent.com/921609/101538645-542bd080-3995-11eb-9fa4-42e14282acd8.png)

**Comment by OliverJAsh** [comment]
@a-tarasyuk Did you manage to reproduce?

**Comment by OliverJAsh** [comment]
I've just tried to write a failing Fourslash test for this but I haven't had any success. However I can still reproduce in VS Code, with no extensions and the latest TS version.

One thing I noticed: it only reproduces when the file has a new line at the very end, e.g. this doesn't reproduce:

![image](https://user-images.githubusercontent.com/921609/119198778-d1e62300-ba81-11eb-84f5-afeadd4fb7a5.png)

but this does:

![image](https://user-images.githubusercontent.com/921609/119198750-c2ff7080-ba81-11eb-9856-345a224e24a3.png)

Notice line 6 is an empty new line.

@a-tarasyuk Are you able to reproduce when you have a new line at the end of the file?

**Comment by zardoy** [comment]
I want to help resolve this one.
@OliverJAsh do you think this might be related to https://github.com/microsoft/TypeScript/issues/53120 (incorrect offsets from refactorings on empty newline)? This is really vscode-specific, I think need to debug vscode extension instead.

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
